compiler/modules/front/codeGen.bal

```diff
@@ -607,7 +667,7 @@ function mustLookup(CodeGenContext cx, string name, Scope? scope, boolean forAss
         return cx.semanticErr(`variable ${name} not found`);
     }
     else if forAssign && binding.isFinal {
-        return cx.semanticErr(`cannot assign to parameter ${name}`);
+        return cx.semanticErr(`cannot assign to final variable ${name}`);
```

Comment:
How does this deal with the parameter case?

Reply:
Both assignment to loop var and assignment to parameter come here. Currently way to distinguish.